tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -616,6 +635,151 @@ public void listBucketZeroMaxKeys() throws Exception {
+
+    final Map<String, String> parameters = new HashMap<>();
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    //parameters with prefix=""
+    expected = new ListBucketResult("bucket", statuses,
+        ListBucketOptions.defaults().setPrefix(""));
+    assertEquals(0, expected.getContents().size());
+
+    parameters.put("prefix", "");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    // Test ListObjects with objects in the bucket
+    mFileSystem.createFile(new AlluxioURI("/bucket/file0"));
+    mFileSystem.createDirectory(new AlluxioURI("/bucket/file_store"));
+    mFileSystem.createFile(new AlluxioURI("/bucket/file_store/file1"));
+    statuses = mFileSystem.listStatus(new AlluxioURI("/bucket"),
+        ListStatusPOptions.newBuilder().setRecursive(true).build());
+
+    //parameters with no prefix
+    expected = new ListBucketResult("bucket", statuses,
+        ListBucketOptions.defaults());
+    assertEquals(3, expected.getContents().size());
+    assertEquals("file0", expected.getContents().get(0).getKey());
+    assertEquals("file_store/", expected.getContents().get(1).getKey());
+    assertEquals("file_store/file1", expected.getContents().get(2).getKey());
+
+    parameters.remove("prefix");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    //parameters with prefix=""
+    expected = new ListBucketResult("bucket", statuses,
+        ListBucketOptions.defaults().setPrefix(""));
+    assertEquals(3, expected.getContents().size());
+    assertEquals("file0", expected.getContents().get(0).getKey());
+    assertEquals("file_store/", expected.getContents().get(1).getKey());
+    assertEquals("file_store/file1", expected.getContents().get(2).getKey());
+
+    parameters.put("prefix", "");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    //parameters with non-existent prefix="dne_folder/file"
+    try {
+      expected = new ListBucketResult("bucket", statuses,
+          ListBucketOptions.defaults().setPrefix("dne_folder/file"));
+    } catch (Exception e) {
+      // expected
+      // TODO(czhu): with the current implementation of prefixes w/o delimiters, there is
+      // never a FileDoesNotExistException because we just list the entire bucket recursively
+      statuses = new ArrayList<>();
+      return;
+    }
```

Comment:
FYI where I'm going with this is that this should throw a `FileDoesNotExistException` if the parent folder also DNE, but doesn't _for now_ because we don't do that optimization on listing files w/o delimiters yet.